source/sessions/tests/dirty-session-errors.yml

```diff
@@ -0,0 +1,281 @@
+runOn:
+  - minServerVersion: "4.0"
+    topology: ["replicaset"]
+  - minServerVersion: "4.1.8"
+    topology: ["sharded"]
+
+database_name: &database_name "session-tests"
+collection_name: &collection_name "test"
+
+data:
+  - {_id: 1}
+
+tests:
+  - description: Clean explicit session is not discarded
+
+    operations:
+      - name: assertSessionNotDirty
+        object: testRunner
+        arguments:
+          session: session0
+      - name: insertOne
+        object: collection
+        arguments:
+          session: session0
+          document: {_id: 2}
+        result:
+          insertedId: 2
+      - name: assertSessionNotDirty
+        object: testRunner
+        arguments:
+          session: session0
+      - name: endSession
+        object: session0
+      - &find_with_implicit_session
+        name: find
+        object: collection
+        arguments:
+          filter: {_id: -1}
+        result: []
+      - name: assertSameLsidOnLastTwoCommands
```

Comment:
Is this redundant? The command started events contain the actual lsid reference (session0) that is expected for each event, so that seems enough of a check

Reply:
Yes it's redundant but I included it for clarity.